pkg/kotsadm/license.go

```diff
@@ -0,0 +1,67 @@
+package kotsadm
+
+import (
+	"bytes"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/kots/pkg/kotsadm/types"
+	corev1 "k8s.io/api/core/v1"
+	kuberneteserrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+func getLicenseSecretYAML(deployOptions *types.DeployOptions) (map[string][]byte, error) {
+	docs := map[string][]byte{}
+	s := json.NewYAMLSerializer(json.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
+
+	var b bytes.Buffer
+	if err := s.Encode(deployOptions.License, &b); err != nil {
+		return nil, errors.Wrap(err, "failed to encode license")
+	}
+
+	var license bytes.Buffer
+	if err := s.Encode(licenseSecret(deployOptions.Namespace, b.String()), &license); err != nil {
+		return nil, errors.Wrap(err, "failed to marshal license secret")
+	}
+	docs["secret-license.yaml"] = license.Bytes()
+
+	return docs, nil
+}
+
+func ensureLicenseSecret(deployOptions *types.DeployOptions, clientset *kubernetes.Clientset) error {
+	existingSecret, err := getLicenseSecret(deployOptions.Namespace, clientset)
+	if err != nil {
+		return errors.Wrap(err, "failed to check for existing license secret")
+	}
+
+	if existingSecret == nil {
```

Comment:
Should this always replace the secret?  What if the user tries this a second time (for whatever reason) with a different license.

Reply:
@divolgin I think this should remove the secret. If the user puts a different license in (and restarts the kotsadm pod), it will create a secret again and import the app, and then delete that secret.  Unless I'm not understanding the concern?